src/main/java/org/broadinstitute/hellbender/utils/genotyper/UnfilledReadsLikelihoods.java

```diff
@@ -0,0 +1,123 @@
+package org.broadinstitute.hellbender.utils.genotyper;
+
+import htsjdk.samtools.util.Locatable;
+import htsjdk.variant.variantcontext.Allele;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import org.broadinstitute.hellbender.utils.IndexRange;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.pileup.PileupElement;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+//TODO comment this off
+public class UnfilledReadsLikelihoods<A extends Allele> extends ReadLikelihoods<A> {
+
+    private Map<String, List<PileupElement>> stratifiedPileups;
+
+    /**
+     * Constructs a new read-likelihood collection.
+     * <p>
+     * <p>
+     * The initial likelihoods for all allele-read combinations are
+     * 0.
+     * </p>
+     *
+     * @param samples all supported samples in the collection.
+     * @param alleles all supported alleles in the collection.
+     * @param reads   reads stratified per sample.
+     * @throws IllegalArgumentException if any of {@code allele}, {@code samples}
+     *                                  or {@code reads} is {@code null},
+     *                                  or if they contain null values.
+     */
+    public UnfilledReadsLikelihoods(SampleList samples, AlleleList<A> alleles, Map<String, List<GATKRead>> reads) {
```

Comment:
I suspect there are a lot of other methods on this class that should be overriden to throw.  I feel like this whole thing is kind of problematic and doesn't result in clean code elsewhere.  I don't have a better design in my head, since somewhere we essentially need a switch that does totally different things with totally different input data.  Maybe the right thing is to have 2 different methods in annotate that accept different data?  We have to get this in soon though so maybe we should revisit.